controllers/ocirepository_controller_test.go

```diff
@@ -80,13 +80,15 @@ func TestOCIRepository_Reconcile(t *testing.T) {
 		tag            string
 		semver         string
 		digest         string
+		mediaType      string
 		assertArtifact []artifactFixture
 	}{
 		{
-			name:   "public tag",
-			url:    podinfoVersions["6.1.6"].url,
-			tag:    podinfoVersions["6.1.6"].tag,
-			digest: podinfoVersions["6.1.6"].digest.Hex,
+			name:      "public tag",
+			url:       podinfoVersions["6.1.6"].url,
+			tag:       podinfoVersions["6.1.6"].tag,
+			digest:    podinfoVersions["6.1.6"].digest.Hex,
+			mediaType: "application/vnd.docker.image.rootfs.diff.tar.gzip",
```

Comment:
It'd be great to test some edge cases here, too.

1. no layer matches the given media type
2. more than layer matches

Reply:
I've added tests for default media type, CLI media type and unknown type.